framework/classes/fuel/orm/model.php

```diff
@@ -385,32 +385,47 @@ public static function behaviours($specific = null, $default = null)
         }
 
         if ($specific) {
-            $behaviours = static::$_behaviours_cached[$class];
-
-            // The $specific argument is not dotated and may contain some keys to return in the behaviour config
-            $specific_fragments = explode('.', $specific, 2);
-            $specific_class = $specific_fragments[0];
-            $specific_key = \Arr::get($specific_fragments, 1, null);
-
-            // To keep backward-compatibility, we have to handle the case where a behaviour exists multiple times in the same array,
-            // with one as a sub-class of $specific, and the other as the exact required class
-            if (isset($behaviours[$specific_class])) {
-                return \Arr::get($behaviours, $specific, $default);
-            }
-
-            // We have to test every key since the required behaviour may be declared through a sub-class
-            foreach ($behaviours as $behaviour_class => $behaviour) {
-                if (is_a($behaviour_class, $specific_class, true)) {
-                    return empty($specific_key) ? $behaviour : \Arr::get($behaviour, $specific_key, $default);
+            if (is_array($specific)) {
+                $return = array();
+                foreach ($specific as $behaviour) {
+                    $return[$behaviour] = self::findChildBehaviour($behaviour, $default);
                 }
+            } else {
+                $return = self::findChildBehaviour($specific, $default);
             }
 
-            return $default;
+            return $return;
         }
 
         return static::$_behaviours_cached[$class];
     }
 
+    protected static function findChildBehaviour($specific, $default)
+    {
+        $class = get_called_class();
```

Comment:
Will this return `Nos\Orm\Model` instead of the actual model ? Should me moved to an argument